### PR TITLE
Refs #10055 - Initial model bindings for OSTREE CRUD

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Comment by parthaa** [comment]
[test]

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -247,9 +252,25 @@ def find_gpg_key
...
+      if params[:action] == "create"
+        params.require(:repository).permit(:label,
+                                           :content_type,
+                                           :url,
+                                           :gpg_key_id,
+                                           :unprotected,
+                                           :name,
+                                           :checksum_type,
+                                           :docker_upstream_name,
+                                           :ostree_branches => [])
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Why did you break apart the array here into two separate instances of calling `permit`?

Comment (ehelms) [inline app/controllers/katello/api/v2/repositories_controller.rb]: I have also never understood the style of dangling when its method calls, you wouldn't write a hash declaration that way, e.g.:

```
params.require(:repository).permit({:label => yo,
                                           :content_type => blah,
                                           :url => stuff,
                                           :gpg_key_id => foo,
                                           :docker_upstream_name => things,
                                           :ostree_branches => []})
```

vs.

```
params.require(:repository).permit(
  :label,
  :content_type,
  :url,
  :docker_upstream_name,
  :ostree_branches => []
)
```

/endstylerant

Comment (parthaa) [inline app/controllers/katello/api/v2/repositories_controller.rb]: > Why did you break apart the array here? I think having two copies of the array is harder to maintain, no?

I didn't realize I could have an array that looked like this 
```[:url, :gpg_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name, :ostree_branches => []]```
where its symbols and last arg being an Hash. Been corrected now. Thanks for pointing out

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
APJ

**Pull request merged**